### ISSUE TITLE
Template variable warning almost 1000 times in the last hour

### ISSUE THREAD

**Issue by ForTheRush** [SgtBatten/HA_blueprints#272]
### Description

Seems to be happening any time it detects motion and goes into a debug loop which is apparently thousands of times a day 

### Version

0.12.0.4c

### Automation Config

```yaml
alias: Frigate notifications test 3
description: ""
use_blueprint:
  path: SgtBatten/Frigate Camera Notifications/Beta.yaml
  input:
    camera: camera.front_door_camera
    notify_device: 3e468c9f3d5acd51229761ae0a8a2434
    base_url: https://xxxxxxxxxxxx.ui.nabu.casa
    title: Alert 🚨
    message: A Person is detected at the front door
    update_sub_label: false
    attachment: snapshot.jpg
    video: >-
      {{base_url}}/api/frigate{{client_id}}/notifications/{{id}}/{{camera}}/clip.mp4
    volume: 0
    sticky: true
    channel: alarm_stream
    zones:
      - stairs
    labels:
      - person
      - cat
      - package
    cooldown: 0
    silence_timer: 0
    button_3: "View stream "
    url_3: >-
      {{base_url}}/api/camera_proxy_stream/camera.{{trigger.payload_json['after']['camera']
      | lower | replace('-','_')}}?token={{state_attr( 'camera.' ~ camera,
      'access_token'
```

### Frigate Config

```yaml
mqtt:
  host: 192.168.1.9
  port: 1883
  user: mqttuser
  password: mqttpass
  topic_prefix: frigate
  client_id: frigate

tls:
  enabled: false

environment_vars:
  LIBVA_DRIVER_NAME: radeonsi

ffmpeg:
  hwaccel_args: preset-vaapi

detectors:
  coral_usb:
    type: edgetpu
    device: usb
  coral_pci1:
    type: edgetpu
    device: pci:0

model:
  path: plus://ca6117fdff32ca1a708cfea1265f69b0

database:
  path: /config/frigate.db

objects:
  track:
    - person
    - cat
    - car
    - amazon
    - fedEx
    - ups
    - package

snapshots:
  enabled: true
  clean_copy: true
  timestamp: false
  bounding_box: true
  crop: false
  retain:
    default: 2
    objects:
      person: 3
      cat: 2
  quality: 100

detect:
  enabled: true
  fps: 10
  min_initialized: 5
  max_disappeared: 30
  stationary:
    interval: 50
    threshold: 50


motion:
  # Optional: The threshold passed to cv2.threshold to determine if a pixel is different enough to be counted as motion. (default: shown below)
  # Increasing this value will make motion detection less sensitive and decreasing it will make motion detection more sensitive.
  # The value should be between 1 and 255.
  threshold: 80

go2rtc:
  streams:
    front_door_camera_main:
      - rtsp://frigateuser:frigatePassword@192.168.1.178:554/h264Preview_01_main
      - ffmpeg:front_door_camera_main#audio=opus
    front_door_camera_sub:
      - rtsp://frigateuser:frigatePassword@192.168.1.178:554/h264Preview_01_sub
      - ffmpeg:front_door_camera_sub#audio=opus
    car_view_camera_main:
      - rtsp://frigateuser:frigatePassword@192.168.1.210:554/h264Preview_01_main
      - ffmpeg:car_view_camera_main#audio=opus
    car_view_camera_sub:
      - rtsp://frigateuser:frigatePassword@192.168.1.210:554/h264Preview_01_sub
      - ffmpeg:car_view_camera_sub#audio=opus
    court_camera_main:
      - rtsp://frigateuser:frigatepassword@192.168.1.123/stream1
      - ffmpeg:court_camera_main#audio=opus
    court_camera_sub:
      - rtsp://frigateuser:frigatepassword@192.168.1.123/stream2
      - ffmpeg:court_camera_sub#audio=opus
  webrtc:
    candidates:
      - stun:8555  # if you have dynamic public IP-address
  hass:
    config: /config

cameras:
  front_door_camera:
    ffmpeg:
      output_args:
        record: preset-record-generic-audio-copy
      inputs:
        - path: rtsp://192.168.1.208:8554/front_door_camera_main
          input_args: preset-rtsp-restream
          roles:
            - record
            - audio
        - path: rtsp://127.0.0.1:8554/front_door_camera_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
    live:
      stream_name: front_door_camera_main
    record:
      enabled: true
      retain:
        days: 2
        mode: active_objects
      events:
        retain:
          default: 1
          mode: active_objects
          objects:
            person: 5
            car: 0.5
            cat: 5

    detect:
      width: 640
      height: 480
      enabled: true

    audio:
      enabled: true
    ui:
      order: 0

    zones:
      stairs:
        coordinates: 
          1,1,0.875,0.769,0.83,0.812,0.832,0.584,0.913,0.598,0.902,0.36,0.651,0.393,0.645,0.121,0.32,0.113,0.319,1,0.593,1
        inertia: 3
        objects:
          - cat
          - fedEx
          - person
          - amazon
          - package
    motion:
      mask:
        - 0.064,0.269,0.081,0.887,0.287,0.953,0.248,0.614,0.28,0.19
        - 0.693,0,0.695,0.077,0.268,0.072,0.266,0
        - 0.833,0.588,0.903,0.6,0.875,0.783,0.871,0.771,0.829,0.811
      threshold: 42
      contour_area: 10
      improve_contrast: false
    review:
      alerts:
        required_zones: stairs
        labels:
          - person
          - package
          - cat
      detections:
        required_zones: stairs
        labels:
          - person
          - cat
          - person
    objects:
      filters:
        car:
          mask: 0.085,0.648,0.265,0.632,0.272,0.896,0.117,0.885
  car_view_camera:
    ffmpeg:
      output_args:
        record: preset-record-generic-audio-copy
      inputs:
        - path: rtsp://192.168.1.208:8554/car_view_camera_main
          input_args: preset-rtsp-restream
          roles:
            - record
            - audio
        - path: rtsp://127.0.0.1:8554/car_view_camera_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
    record:
      enabled: true
      retain:
        days: 0
      events:
        retain:
          default: 1
          mode: active_objects
          objects:
            car: 3
            person: 3
    detect:
      width: 640
      height: 360
      enabled: true

    audio:
      enabled: false
    live:
      stream_name: car_view_camera_main
    zones:
      parking:
        coordinates: 0.057,0.323,0.1,0.425,0.904,0.509,0.925,0.409
        inertia: 3
        loitering_time: 5
        objects:
          - car
          - person
      dannys_spot:
        coordinates: 0.382,0.347,0.404,0.237,0.587,0.264,0.581,0.369
        objects:
          - car
          - person
        inertia: 3
      spot2:
        coordinates: 0.292,0.223,0.405,0.228,0.383,0.35,0.255,0.336
        objects:
          - car
          - person
        inertia: 3
      kays_spot:
        coordinates: 0.07,0.322,0.134,0.216,0.282,0.231,0.267,0.335
        inertia: 3
        objects:
          - car
          - person
      other_spots:
        coordinates: 0.581,0.368,0.908,0.433,0.92,0.328,0.596,0.256
        objects:
          - car
          - person
        inertia: 3
    motion:
      threshold: 32
      contour_area: 14
      improve_contrast: true
    ui:
      order: 1
    review:
      detections:
        required_zones:
          - spot2
          - kays_spot
          - dannys_spot
          - parking
      alerts:
        required_zones:
          - parking
          - kays_spot
          - dannys_spot
  court_camera:
    ffmpeg:
      output_args:
        record: preset-record-generic-audio-copy
      inputs:
        - path: rtsp://192.168.1.208:8554/court_camera_main?mp4
          input_args: preset-rtsp-restream
          roles:
            - audio
            - record
        - path: rtsp://127.0.0.1:8554/court_camera_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
    record:
      enabled: true
      retain:
        days: 0
        mode: active_objects
      events:
        retain:
          default: 0
          mode: active_objects
          objects:
            cat: 3
    detect:
      width: 640
      height: 480
      enabled: true

    audio:
      enabled: false
    ui:
      order: 2
    live:
      stream_name: court_camera_main
    zones:
      bed:
        coordinates: 0.99,0.541,1,0.384,0.913,0.376,0.805,0.536,0.609,0.611,0.685,1,1,1,0.99,0.75
        inertia: 3
        objects: cat
      bedroom_door:
        coordinates: 0.356,0.65,0.438,0.778,0.424,0,0.365,0
        inertia: 3
        objects: cat
        loitering_time: 2
      cat_tree:
        coordinates: 0.206,0.056,0.365,0.063,0.375,0.651,0.319,0.635,0.313,0.49,0.207,0.45
        inertia: 3
        objects: cat
      in_there:
        coordinates: 0.435,0.772,0.488,0.911,0.655,0.914,0.535,0,0.424,0
        objects: cat
        inertia: 3
      closet:
        coordinates: 0.605,0.605,0.789,0.549,0.911,0.366,1,0.409,1,0.193,0.764,0.168,0.559,0.157
        objects:
          - person
          - cat
        inertia: 3
    motion:
      threshold: 61
      contour_area: 25
      improve_contrast: 'false'
      mask:
        - 0.519,0,0.569,0.159,1,0.191,1,0
        - 0,1,0.515,1,0.497,0.92,0.367,0.665,0.295,0.655,0.228,0.496,0.202,0.466,0.199,0,0,0
        - 0,0.044,0.358,0.053,0.37,0,0,0
    review:
      detections:
        labels:
          - cat
          - person
        required_zones:
          - bedroom_door
          - in_there
      alerts:
        required_zones:
          - in_there
          - bedroom_door
        labels:
          - person
          - cat

    objects:
      filters:
        car:
          mask: 0.454,0.845,0.125,0.895,0.132,1,0.504,1

        person:
          mask: 0.077,0.457,0.129,0.462,0.121,0.398,0.078,0.395
      mask: 0.349,0.835,0.387,0.858,0.375,0.78,0.334,0.769
version: 0.14
camera_groups:
  my_cams:
    order: 1
    icon: LuAlignHorizontalDistributeStart
    cameras:
      - car_view_camera
      - court_camera
      - front_door_camera
logger:
  logs:
    frigate.record.maintainer: debug
```

### Any other relevant information

```yaml
Logger: homeassistant.helpers.template
Source: helpers/template.py:2651
First occurred: August 31, 2024 at 11:15:14 PM (942 occurrences)
Last logged: 12:36:13 AM

Template variable warning: 'zones_satisified' is undefined when rendering 'DEBUG (in loop): Send Notification: {{ loitering or (custom_filter and not home and zones_satisified and state_true and (new_snapshot or presence_changed or stationary_moved or entered_new_zones or entered_new_filter_zones or sub_label_changed)) }} Info: sublabel: {{sub_label}}, image: "{{base_url}}/api/frigate{{client_id}}/notifications/{{id}}/{{attachment}}" Title: {{title}} message: {{message}} iOS sound: {{'Critical' if critical else 'disabled by alert once' if alert_once else 'enabled'}}, Android Sound: {{'disabled by alert once' if alert_once else 'enabled'}}, iOS url: /api/frigate{{client_id}}/notifications/{{id}}/{{camera + '/clip.mp4' if video|length>0 and wait.trigger.payload_json['type'] == 'end' else attachment }} video: "{{video}}" critical: {{critical}}, tts: {{tts}} Triggers: New Snapshot: {{'True' if event['before']['snapshot']['frame_time'] != event['after']['snapshot']['frame_time'] else 'False'}}{{' - Disabled' if not update_thumbnail}} Presence Changed: {{presence_changed}}, Stationary Moved: {{stationary_moved}}, Entered Zones Changed: {{'True' if event['before']['entered_zones'] != event['after']['entered_zones'] else 'False'}}{{' - Disabled' if not zone_only}}, sublabel changed: {{ sub_label != sub_label_before }}{{' - Disabled' if not update_sub_label}}, Conditions: Loitering: {{loitering}} or Filters: Zones: Zone Filter toggle on: {{zone_only}}, Multi-Zone toggle on: {{zone_multi}}, Required zones: {{input_zones}}, Zone Order toggle on: {{zone_order_enforced}} Last Zones: {{last_zones}}, Entered Zones: {{ entered_zones | list | length }} - {{entered_zones}}, TEST: {{'PASS' if zones_satisfied else 'FAIL' }} {{'(Multi)' if zone_only and zone_multi}} {{'(Order-Enforced)' if zone_only and zone_multi and zone_order_enforced}}, Object Filter: Input: {{input_labels}}, TEST: {{'PASS' if not labels|length or object in labels else 'FAIL'}}, Presence entity (not home): Entity: {{presence_entity}}, TEST: {{'PASS' if not home else 'FAIL'}}, Time Filter: Disabled times: {{disable_times}}, TEST: {{'PASS' if now().hour not in disable_times else 'FAIL'}} State Filter: State Filter toggle on: {{state_only}}, State Filter Entity: {{input_entity}}, Required States: {{input_states}}, TEST: {{'PASS' if state_true else 'FAIL'}}, Custom Filter: {{'PASS' if custom_filter else 'FAIL'}},'
```

**Comment by SgtBatten** [comment]
Please try with the latest version. c is several updates old now

**Comment by ForTheRush** [comment]
I've updated per your recommendation. Instead of 1000s an hour now it's only about 15-30 occurrences an hour which is definitely better but still all up in logs every day

**Comment by SgtBatten** [comment]
Can you share the template error front he logs. I may have missed one

**Comment by ForTheRush** [comment]
`Logger: homeassistant.helpers.template
Source: helpers/template.py:2667
First occurred: 5:01:48 AM (165 occurrences)
Last logged: 9:25:16 AM

Template variable warning: 'zones_satisified' is undefined when rendering 'DEBUG (in loop): Send Notification: {{ loitering or (custom_filter and not home and zones_satisified and state_true and (new_snapshot or presence_changed or stationary_moved or entered_new_zones or entered_new_filter_zones or sub_label_changed)) }} Info: sublabel: {{sub_label}}, image: "{{base_url}}/api/frigate{{client_id}}/notifications/{{id}}/{{attachment}}" Title: {{title}} message: {{message}} iOS sound: {{'Critical' if critical else 'disabled by alert once' if alert_once else 'enabled'}}, Android Sound: {{'disabled by alert once' if alert_once else 'enabled'}}, iOS url: /api/frigate{{client_id}}/notifications/{{id}}/{{camera + '/clip.mp4' if video|length>0 and wait.trigger.payload_json['type'] == 'end' else attachment }} video: "{{video}}" critical: {{critical}}, tts: {{tts}} Triggers: New Snapshot: {{'True' if event['before']['snapshot']['frame_time'] != event['after']['snapshot']['frame_time'] else 'False'}}{{' - Disabled' if not update_thumbnail}} Presence Changed: {{presence_changed}}, Stationary Moved: {{stationary_moved}}, Entered Zones Changed: {{'True' if event['before']['entered_zones'] != event['after']['entered_zones'] else 'False'}}{{' - Disabled' if not zone_only}}, sublabel changed: {{ sub_label != sub_label_before }}{{' - Disabled' if not update_sub_label}}, Conditions: Loitering: {{loitering}} or Filters: Zones: Zone Filter toggle on: {{zone_only}}, Multi-Zone toggle on: {{zone_multi}}, Required zones: {{input_zones}}, Zone Order toggle on: {{zone_order_enforced}} Last Zones: {{last_zones}}, Entered Zones: {{ entered_zones | list | length }} - {{entered_zones}}, TEST: {{'PASS' if zones_satisfied else 'FAIL' }} {{'(Multi)' if zone_only and zone_multi}} {{'(Order-Enforced)' if zone_only and zone_multi and zone_order_enforced}}, Object Filter: Input: {{input_labels}}, TEST: {{'PASS' if not labels|length or object in labels else 'FAIL'}}, Presence entity (not home): Entity: {{presence_entity}}, TEST: {{'PASS' if not home else 'FAIL'}}, Time Filter: Disabled times: {{disable_times}}, TEST: {{'PASS' if now().hour not in disable_times else 'FAIL'}} State Filter: State Filter toggle on: {{state_only}}, State Filter Entity: {{input_entity}}, Required States: {{input_states}}, TEST: {{'PASS' if state_true else 'FAIL'}}, Custom Filter: {{'PASS' if custom_filter else 'FAIL'}},'`

**Comment by SgtBatten** [comment]
What version are you now using?

**Comment by ForTheRush** [comment]
For the log above I was using d I had tried e for a few days but it was the same. At least d is far less errors on my system

**Comment by SgtBatten** [comment]
I thought i'd fixed it but apprently i can't spell. 

fixed in 0.12.0.4j
